softwerkskammer/lib/authentication/index.js

```diff
@@ -113,6 +119,48 @@ function setupGooglePlus(app1) {
   }
 }
 
+function setupMagicLink(app1) {
+
+  const strategy = new MagicLinkStrategy({
+      secret: conf.get('magicLinkSecret'),
+      tokenName: 'token',
+      tokenProblemRedirect: '/' // 'tokenProblem.html'
+    },
+    authenticationService.createUserObjectFromMagicLink
+  );
+
+  passport.use(strategy);
+  createProviderAuthenticationRoutes(app1, strategy.name);
+
+  app1.get('/magiclinkmail', (req, res, next) => {
```

Comment:
Ich würde es bevorzugen, wenn die routenbehandlung außerhalb der strategy passiert

Reply:
Ich wollte Konsistenz herstellen zu den anderen Fällen. Da findet die Routenbehandlung ja auch innerhalb der Strategy-erstellenden Funktion statt, nämlich durch den Aufruf von createProviderAuthenticationRoutes(). Sieht man halt nur nicht so deutlich...

Zukunftspläne:
Vielleicht lässt sich auch dieser Teil in die Strategy selbst verlagern, den Aspekt habe ich noch nicht so durchschaut. Denn andere Strategien sind ja auch zweiteilig.

Reply:
OK